### ISSUE TITLE
Non-existent select or update property not causing errors in some cases

### ISSUE THREAD

**Issue by Arlen22** [prisma/prisma#20032]
### Bug description

I'm running into a bug where including nonexistent fields in `update({ data: {...here} })` does not throw an error. However, adding it to the select section (`update({ data: {...here}, select: {...here} })`) causes both (data and select) to show the expected errors. 

In a `findUnique({ where: {}, select: { ...here } })` statement, the extra property is listed as a `never` type, but then when written out, the type is `undefined`

### How to reproduce

I've renamed a couple fields for privacy, but the table names are unchanged. I included the return type as it seemed relevant. The code in the Prisma Information section actually has this problem for me, as described. I first made this post with edited code that had the problem, but now I actually compiled the code below and verified that it has the problem. 

- Paste the prisma schema into a schema.prisma file and run `prisma generate`
- Paste the code into a typescript file beside the prisma file. 
- Profit! Er, verify there is no typescript error thrown by `tsc ./file.ts`.

### Expected behavior

I expect an error on the line `stora: true,`

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["jsonProtocol"]
  output          = "./client"
  binaryTargets   = ["native", "debian-openssl-1.0.x"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

enum DocusignEnvelopeStatus {
  none
}

model Customer {
  id                    String                 @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  createdAt             DateTime               @default(now())
  updatedAt             DateTime               @updatedAt
  /// [any]
  extra                 Json?
  WelcomeEmailCompleted Boolean?
  DocusignCompleted     Boolean?
  DocusignStatus        DocusignEnvelopeStatus @default(none)
  DocusignEnvelopeID    String?                @unique @db.Uuid
  billingID             String                 @unique @db.Uuid
  /// [any]
  WelcomeEmailInfo      Json?
}

model Rental {
  id        String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  /// [any]
  extra     Json?
  StartDate String?
}

```

```ts
// Add your code using Prisma Client
import * as PrismaExtra from "./client";
import { PrismaClient } from "./client"
import { DefaultArgs } from "./client/runtime";
declare global {
  namespace PrismaJson {
    export type Types = any;
  }
}
const inner = new PrismaClient<{
  datasources: {
    db: {
      url: string;
    };
  };
  log: "error"[];
}, never, false, DefaultArgs>();
const client = inner.$extends({
  result: {
    rental: {
      MyDay: {
        needs: { StartDate: true },
        compute: ({ StartDate }) => StartDate ? new Date().getDate() : null,
      }
    }
  },
  query: {
    $allOperations: async ({ args, query }) => {
      try {
        //other code
        return query(args);
      } finally {
        //other code
      }
    }
  }
});
async function test() {

  // type PrismaClientType = PrismaConnection["client"];
  // type PrismaTxnClient = Omit<PrismaClientType, "$on" | "$connect" | "$disconnect" | "$use" | "$transaction">;
  const customer = await client.customer.findUnique({
    where: { id: "the standard uuid lookup" },
    select: {
      billingID: true,
      DocusignEnvelopeID: true,
      DocusignStatus: true,
      DocusignCompleted: true,
      stora: true, // random nonsense that doesn't cause a typescript error
      WelcomeEmailCompleted: true,
    }
  });
  /** and the return type */
  const result: {
    billingID: string;
    DocusignEnvelopeID: string | null;
    DocusignStatus: PrismaExtra.DocusignEnvelopeStatus;
    DocusignCompleted: boolean | null;
    stora: never;
    WelcomeEmailCompleted: boolean | null;
  } | null = customer;
}
```

### Environment & setup

- OS:  Ubuntu 22 
- Database: Postgresql 
- Node.js version:  v18.6.0

### Prisma Version

```
prisma                  : 4.16.1
@prisma/client          : 4.16.1
Current platform        : debian-openssl-3.0.x
Query Engine (Node-API) : libquery-engine b20ead4d3ab9e78ac112966e242ded703f4a052c (at node_modules/prisma/node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Migration Engine        : migration-engine-cli b20ead4d3ab9e78ac112966e242ded703f4a052c (at node_modules/prisma/node_modules/@prisma/engines/migration-engine-debian-openssl-3.0.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.16.0-66.b20ead4d3ab9e78ac112966e242ded703f4a052c
Default Engines Hash    : b20ead4d3ab9e78ac112966e242ded703f4a052c
Studio                  : 0.484.0
Preview Features        : jsonProtocol

```

**Comment by Arlen22** [comment]
Gah! Appears to be fixed in `4.16.2`. It correctly throws an error. I'll leave this open for the moment in case anyone else comes around with the same issue.

**Comment by Jolg42** [comment]
I think this was fixed related to https://github.com/prisma/prisma/issues/19958 in https://github.com/prisma/prisma/pull/19942

Closing this as [`4.16.2`](https://github.com/prisma/prisma/releases/tag/4.16.2) is now available.
